openedx/core/djangoapps/notifications/views.py

```diff
@@ -130,3 +139,133 @@ def patch(self, request, course_key_string):
+        """
+        queryset = super().get_queryset()
+        app_name = self.request.query_params.get('app_name')
+        if app_name:
+            queryset = queryset.filter(app_name=app_name, user=self.request.user)
+        return queryset
+
+
+class NotificationCountViewSet(viewsets.ViewSet):
+    """
+    API view for getting the unseen notifications count for a user.
+    """
+
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request):
+        """
+        Get the unseen notifications count for a user.
+
+        **Permissions**: User must be authenticated.
+        **Response Format**:
+        ```json
+        {
+            "count": (int) total_number_of_unseen_notifications,
+            "count_by_app_name": {
+                (str) app_name: (int) number_of_unseen_notifications,
+                ...
+            }
+        }
+        ```
+        **Response Error Codes**:
+        - 403: The requester cannot access resource.
+        """
+        # Get the unseen notifications count for each app name.
+        count_by_app_name = (
+            Notification.objects
+            .filter(user_id=request.user, last_seen__isnull=True)
+            .values('app_name')
+            .annotate(count=Count('*'))
+        )
+        count_total = 0
+        count_by_app_name_dict = {}
+
+        for item in count_by_app_name:
+            app_name = item['app_name']
+            count = item['count']
+
+            count_total += count
+            count_by_app_name_dict[app_name] = count
+        # Return the unseen notifications count for the user and the unseen notifications count for each app name.
+
+        return Response({
+            "count": count_total,
+            "count_by_app_name": count_by_app_name_dict,
+        })
+
+
+class MarkNotificationsUnseenAPIView(UpdateAPIView):
+    """
+    API view for marking user's all notifications unseen for a provided app_name.
+    """
+
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def update(self, request, *args, **kwargs):
+        """
+        Marks all notifications for the given app name unseen for the authenticated user.
+
+        **Args:**
+            app_name: The name of the app to mark notifications unseen for.
+        **Response Format:**
+            A `Response` object with a 200 OK status code if the notifications were successfully marked unseen.
+        **Response Error Codes**:
+        - 400: Bad Request status code if the app name is invalid.
+        """
+        app_name = self.kwargs.get('app_name')
+
+        if not app_name:
+            return Response({'message': 'Invalid app name.'}, status=400)
+
+        notifications = Notification.objects.filter(
+            user=request.user,
+            app_name=app_name,
+            last_seen__isnull=True,
+        )
+
+        notifications.update(last_seen=datetime.now())
```

Comment:
Might be worth checking if notifications are not empty?

Reply:
No need to check it `update()` won't do anything if no records are available.